include/alp/omp/storage.hpp

```diff
@@ -403,9 +417,8 @@ namespace alp {
 					const typename Distribution::GlobalCoord global( imf_r.map( i ), imf_c.map( j ) );
 					const typename Distribution::LocalCoord local = distribution.mapGlobalToLocal( global );
 
-					const size_t thread = local.tr * distribution.getThreadGridDims().second + local.tc;
-
-					const size_t local_block = local.br * distribution.getLocalBlockGridDims( local.tr, local.tc ).second + local.bc;
+					const size_t thread = distribution.getThreadId( local.getThreadCoords() );
+					const size_t local_block = local.br * distribution.getLocalBlockGridDims( local.getThreadCoords() ).second + local.bc;
```

Comment:
If logic for computing block_id goes into the distribution than this could also be adapted accordingly